lib/connection.js

```diff
@@ -0,0 +1,172 @@
+const { dbconn } = require('idb-connector');
+
+/**
+ * @class Connection
+ * @constructor
+ */
+class Connection {
+  constructor(db = { url: '', username: '', password: '' }) {
+    const me = this;
```

Comment:
If we are going to use arrow functions now, we can probably get rid of all this `me/self = this` nonsense.

Reply:
There are times when its needed to save a ref to the current context of this.

Agreed we should probably only save the ref to this when needed

@brianmjerome  any thoughts?

Reply:
@abmusse I still cache `this` for closures and readability. I wouldn't say it's nonsense. Whatever the decision, I think it should be consistent throughout the codebase.

PS: I got 2 git profiles. I'll get more notifications going forward with @bjeromeHCS instead.

Reply:
I took a cursory glance through the code and couldn't find any location where closure would be needed. And I would argue that renaming `this`, a language keyword, to something else actually reduces readability.

Reply:
@markdirish I won't argue your coding methods or how you read it. As long as there is a consensus and the coding is consistent I don't care if it's `this` or `me` or `chicken`.

Reply:
@markdirish technically not renaming `this`, just making a copy, but I agree - from what I can see there's no reason to make a copy in any of the code here, just use `this` directly.

Reply:
@bjeromeHCS haha although chicken is tempting I will go forward using `this` directly and not save of a reference to it unless we need to.

@kadler Agreed

Reply:
Pushed update